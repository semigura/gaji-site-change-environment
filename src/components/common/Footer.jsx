import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'gatsby';
import Icon from './icons/Icon';

import css from './Footer.module.scss';

const Footer = props => {
  const { className, name, facebook, twitter, instagram } = props;
  const classes = classNames(css.Footer, className);
  const icons = [
    {
      href: facebook,
      name: 'facebook',
      title: 'Facebook',
    },
    {
      href: twitter,
      name: 'twitter',
      title: 'Twitter',
    },
    {
      href: instagram,
      name: 'instagram',
      title: 'Instagram',
    },
  ];

  return (
    <footer className={classes}>
      <div className={css.Footer__Links}>
        <div className={css.Footer__LinksInner}>
          <Link className={css.Footer__Logo} to="/">
            <Icon className={css.Footer__Icon} name="logo" />
          </Link>
          <ul className={css.Footer__List}>
            {icons.map(icon => (
              <li className={css.Footer__ListItem} key={icon.name}>
                <a
                  className={css.Footer__SNS}
                  href={icon.href}
                  title={icon.title}
                  rel="nofollow noreferrer"
                >
                  <Icon name={icon.name} />
                </a>
              </li>
            ))}
          </ul>
        </div>
      </div>
      <div className={css.Footer__Footer}>
        <ul className={css.Footer__MenuList}>
          <li>
            <Link className={css.Footer__Link} to="/about/access/">
              アクセス
            </Link>
          </li>
          <li>
            <Link className={css.Footer__Link} to="/privacy/">
              プライバシーポリシー
            </Link>
          </li>
          <li>
            <Link className={css.Footer__Link} to="/recruit/">
              採用情報
            </Link>
          </li>
          <li>
            <Link className={css.Footer__Link} to="/sitemap/">
              サイトマップ
            </Link>
          </li>
        </ul>
        <span className={css.Footer__Copyright}>{`©️ ${name} 2016`}</span>
      </div>
    </footer>
  );
};

Footer.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  facebook: PropTypes.string.isRequired,
  twitter: PropTypes.string.isRequired,
  instagram: PropTypes.string.isRequired,
};

Footer.defaultProps = {
  className: undefined,
};

export default Footer;
