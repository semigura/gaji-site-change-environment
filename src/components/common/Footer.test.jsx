import { shallow } from 'enzyme';
import React from 'react';

import Footer from './Footer';

describe('<Footer />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Footer
        name="Gaji-Labo Inc."
        facebook="https://www.facebook.com/gajilabo/"
        twitter="https://twitter.com/gaji_labo"
        instagram="https://www.instagram.com/gaji_labo/"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
