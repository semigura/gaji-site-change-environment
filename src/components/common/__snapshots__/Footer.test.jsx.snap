// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<footer
  className="Footer"
>
  <div
    className="Footer__Links"
  >
    <div
      className="Footer__LinksInner"
    >
      <mockConstructor
        className="Footer__Logo"
        to="/"
      >
        <Icon
          className="Footer__Icon"
          name="logo"
        />
      </mockConstructor>
      <ul
        className="Footer__List"
      >
        <li
          className="Footer__ListItem"
          key="facebook"
        >
          <a
            className="Footer__SNS"
            href="https://www.facebook.com/gajilabo/"
            rel="nofollow noreferrer"
            title="Facebook"
          >
            <Icon
              name="facebook"
            />
          </a>
        </li>
        <li
          className="Footer__ListItem"
          key="twitter"
        >
          <a
            className="Footer__SNS"
            href="https://twitter.com/gaji_labo"
            rel="nofollow noreferrer"
            title="Twitter"
          >
            <Icon
              name="twitter"
            />
          </a>
        </li>
        <li
          className="Footer__ListItem"
          key="instagram"
        >
          <a
            className="Footer__SNS"
            href="https://www.instagram.com/gaji_labo/"
            rel="nofollow noreferrer"
            title="Instagram"
          >
            <Icon
              name="instagram"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="Footer__Footer"
  >
    <ul
      className="Footer__MenuList"
    >
      <li>
        <mockConstructor
          className="Footer__Link"
          to="/about/access/"
        >
          アクセス
        </mockConstructor>
      </li>
      <li>
        <mockConstructor
          className="Footer__Link"
          to="/privacy/"
        >
          プライバシーポリシー
        </mockConstructor>
      </li>
      <li>
        <mockConstructor
          className="Footer__Link"
          to="/recruit/"
        >
          採用情報
        </mockConstructor>
      </li>
      <li>
        <mockConstructor
          className="Footer__Link"
          to="/sitemap/"
        >
          サイトマップ
        </mockConstructor>
      </li>
    </ul>
    <span
      className="Footer__Copyright"
    >
      ©️ Gaji-Labo Inc. 2016
    </span>
  </div>
</footer>
`;
